prov/psm3/psm3/psm_verbs_mr.c

```diff
@@ -1146,6 +1146,7 @@ void psm2_verbs_free_mr_cache(psm2_mr_cache_t cache)
 			if (mrc->refcount) {
 				_HFI_ERROR("unreleased MR in psm2_verbs_free_mr_cache addr %p len %"PRIu64" access 0x%x\n",
 					mrc->addr, mrc->length, mrc->access);
+				return; // leak the rest, let process exit cleanup
 			}
 			mrc->refcount = 0;
```

Comment:
nitpick:  this assignment becomes redundant now.

Reply:
I can remove in future patch if required, but I think we wanted to keep it in. The return early was only supposed to be a temporary workaround and we did not want to remove this code for now.